### PR TITLE
 fix: allowing members table to select all

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
[ENT-9023](https://2u-internal.atlassian.net/browse/ENT-9023)

Page selected, not all records:
image

Page all records selected: 
image

### CODE REVIEW

**Review by marlonkeating — CHANGES_REQUESTED** [review]
Though the text is corrected when all records are selected, it still seems to only delete the first page of results.

https://github.com/openedx/frontend-app-admin-portal/assets/322346/07923950-3947-49c7-8c65-ecdaad4f25b2

File: src/components/learner-credit-management/members-tab/tests/MembersTab.test.jsx [hunk 1]
```diff
@@ -361,14 +361,16 @@ describe('<BudgetDetailPage />', () => {
...
 
-    // TODO Sorting by enrollment count is currently not supported by the backend
-    // userEvent.click(screen.getByTestId('members-table-enrollments-column-header'));
-    // await waitFor(() => expect(mockFetchEnterpriseGroupMembersTableData).toHaveBeenCalledWith({
-    //   filters: [],
-    //   pageIndex: 0,
-    //   pageSize: 10,
-    //   sortBy: [{ desc: false, id: 'enrollmentCount' }],
-    // }));
+    userEvent.click(screen.getByTestId('members-table-enrollments-column-header'));
```
Comment (alex-sheehan-edx) [inline src/components/learner-credit-management/members-tab/tests/MembersTab.test.jsx:364]: This was done, I just had forgotten to uncomment on a previous ticket.

File: src/components/learner-credit-management/members-tab/GroupMembersCsvDownloadTableAction.jsx [hunk 2]
```diff
@@ -69,6 +70,13 @@ const GroupMembersCsvDownloadTableAction = ({
     });
   };
 
+  let buttonSelectedNumber = 0;
+  if (selectedEmailCount > 0) {
+    buttonSelectedNumber = isEntireTableSelected ? tableInstance.itemCount : selectedEmailCount;
```
Comment (katrinan029) [inline src/components/learner-credit-management/members-tab/GroupMembersCsvDownloadTableAction.jsx:75]: [issue] small missing detail but there should be parenthesis around the selected number of learners in the download button e.g. `Download (12)`

perhaps something like this and update the unit test as such:

```
buttonSelectedNumber = isEntireTableSelected ? `(${tableInstance.itemCount})` : `(${selectedEmailCount})`;
```

File: src/components/learner-credit-management/members-tab/tests/MembersTab.test.jsx [hunk 3]
```diff
@@ -741,7 +751,7 @@ describe('<BudgetDetailPage />', () => {
     const toggleAllRowsSelected = screen.getByTitle('Toggle All Current Page Rows Selected');
     userEvent.click(toggleAllRowsSelected);
 
-    const downloadButton = screen.getByText('Download (1)');
+    const downloadButton = screen.getByText('Download 1');
```
Comment (katrinan029) [inline src/components/learner-credit-management/members-tab/tests/MembersTab.test.jsx:754]: based on the feedback above, this should be updated to `Download (1)` to match figma mocks.

**Review by katrinan029 — CHANGES_REQUESTED** [review]
Tested on stage and the select all is working as expected with the exception of the remove learners. I think we could wait to merge this PR after https://github.com/openedx/edx-enterprise/pull/2114/files gets merged and deployed to stage so that we can test the remove learners functionality

**Review by marlonkeating — APPROVED** [review]

**Review by katrinan029 — APPROVED** [review]

**Pull request merged**